src/main.zig

```diff
@@ -319,11 +331,16 @@ const WlrSurface = struct {
 
         // We want to ignore any exclusive zones set by other surfaces.
         self.wlr_surface.setExclusiveZone(-1);
-        self.wlr_surface.setSize(self.width, self.height);
-        self.wl_surface.setBufferScale(@intCast(self.scale));
+        self.wlr_surface.setSize(self.destination_width, self.destination_height);
+        
+        if (fractional_scale_manager) |_| {
+            self.wl_surface.setBufferScale(1);
+        } else {
+            self.wl_surface.setBufferScale(@intCast(self.scale));
+        }
 
-        // TODO: Make the user set this.
-        self.wlr_surface.setAnchor(.{ .top = true, .left = true });
+        // Anchor to all 4 sides to fill the screen.
+        self.wlr_surface.setAnchor(.{ .top = true, .bottom = true, .left = true, .right = true });
```

Comment:
When I tried this with Hyprland I had issues. Could you verify that current Hyprland works fine with this?